### PR TITLE
[FIX] sale: add salesperson to SO invoice follower

### PR DESCRIPTION
**Pull request by hamzaislam101**
The salesperson is not added to the following list of invoice that is generated from a sales order.

1. Create a sales order by assigning a salesperson.
2. Confirm the order and generate the invoice.

Current Behavior:
The salesperson is not added to the message following list of invoice.

Expected Behavior:
The salesperson should be in the message following list of invoice.

OPW-3439198

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
looks ok to me

in 15.0 the salesperson was subscribed thanks to:

https://github.com/odoo/odoo/blob/1e925285cd4969d3de5745cf7a01f92579078ffb/addons/mail/models/mail_thread.py#L3065
https://github.com/odoo/odoo/blob/1e925285cd4969d3de5745cf7a01f92579078ffb/addons/mail/models/mail_thread.py#L3088-L3090
the code seems to be very similar in 16.0, but the user_id (that is a field related to invoice_user_id in account.move) is no longer in the create value list, so the salesperson doesn't get subscribed.

I think unless it was done by design, I think a simpler fix might be to add the user_id back in the _prepare_invoice method like it was in 15.0 version:

https://github.com/odoo/odoo/blob/d33e18e9be50fd1e5dcafaf27b193ee61fb29c4a/addons/sale/models/sale_order.py#L632

File: addons/sale/models/sale_order.py [hunk 1]
```diff
@@ -1008,6 +1008,7 @@ def _prepare_invoice(self):
             'transaction_ids': [Command.set(self.transaction_ids.ids)],
             'company_id': self.company_id.id,
             'invoice_line_ids': [],
+            'user_id': self.user_id.id
```
Comment (Feyensv) [inline addons/sale/models/sale_order.py:1011]:
```suggestion
            'user_id': self.user_id.id,
```

**Review by Feyensv — COMMENTED** [review]
Can we have a little test please ? :pray:

**Review by Feyensv — APPROVED** [review]
@robodoo delegate+

Thanks for the changes :+1:

**Pull request closed without merging**